File: SisifoNotes.Lib/Models/Task.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SisifoNotes.Lib.Models
{
    public class Task : Note
    {
        public Priority Priority { get; set; }
        public DateTime Deadline { get; set; }
    }

    public enum Priority
    {
        Low,
        Medium,
        High,
        Urgent
    }
}

File: SisifoNotes.Web/wwwroot/Scripts/Models/Task.js
class Task extends Note
{
    constructor(json)
    {
        super(json)
        if (json)
        {
            this.Priority = json.priority;
            this.Deadline = json.deadline;
        }
        else
        {
            this.Priority = [];
            this.Deadline = new Date();
        }
    }
}